### PR TITLE
Implement Categories in 1password extension

### PR DESCRIPTION
**Pull request by selfish**
## Description

This PR adds category sections to the 1password extension, to show items of the same type clustered.

## Type of change

- Improvement for somebody else's extension. Extension author: @khasbilegt 

## Screenshots

Before - Mixed Types:

1p-before

After - Categorized view:

1p-after

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I checked that extension's icon has size of `512x512`

### CODE REVIEW

**Comment by pernielsentikaer** [comment]
Hi @selfish 👋 

It sounds like a good update to me 💪 

@khasbilegt what do you think about this update, and can you give it a quick spin?

**Comment by khasbilegt** [comment]
Hello, @selfish. Thank you for your contribution.

@pernielsentikaer this PR looks good to me.

**Review by pernielsentikaer — APPROVED** [review]
Hi @selfish 👋 

Very nice addition to this already awesome extension 💪 

A suggestion for later could be to add the possibility to hide the type subtitle since for some it could be a bit redundant having it in the group title.

I have 200+ items in one of them, so I like it, but if you only have a few it could be a delightful setting to have to for the UX.

This should be added in a new PR, since I'm going to mark this accepted now 😄

File: extensions/1password7/src/list.tsx [hunk 1]
```diff
@@ -112,6 +129,25 @@ function getIconForCategory(categoryUUID: string) {
   }
 }
 
+function PasswordListCategory(props: { onePasswordMetaItemsCategory: OnePasswordMetaItemsCategory }) {
+  const onePasswordMetaItemsCategory = props.onePasswordMetaItemsCategory;
+  console.log(onePasswordMetaItemsCategory)
```
Comment (pernielsentikaer) [inline extensions/1password7/src/list.tsx:134]:
```suggestion
  console.log(onePasswordMetaItemsCategory)
```
Can we remove this, no need to log secret data to the console 😄

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi again 👋 

Please check the suggestion, when it's done It's ready to get merged 💯

File: extensions/1password7/src/list.tsx (hunk 1 above)
Comment (selfish) [reply extensions/1password7/src/list.tsx]: Whoopsie! Good catch.
Debug code left there.

Removed!

**Comment by selfish** [comment]
Thanks for your review.
I've removed the leftover console.log, and will create a followup-PR to add the option to toggle the redundant subtitle.
Cheers!

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @selfish 👋 

Awesome update, it looks SO good 💯 

I just checked the latest version and can see that we have “unique key prop” issue that we need to have addressed before merging, can you look into it?

```
2021-11-17 15:48:56.917 F  Raycast[781:ab35b] [com.raycast.extensions.1password7:list] Warning: Each child in a list should have a unique "key" prop.

Check the render method of `PasswordList`. See https://reactjs.org/link/warning-keys for more information.
    at null.PasswordListCategory (/Users/pernielsentikaer/Desktop/newnew/extensions/extensions/1password7/src/list.tsx:133:46)
    at null.PasswordList (/Users/pernielsentikaer/Desktop/newnew/extensions/extensions/1password7/src/list.tsx:84:25)
    at null.Command (/Users/pernielsentikaer/Desktop/newnew/extensions/extensions/1password7/src/list.tsx:71:59)
    at ray-navigation-stack
    at nm (/Applications/Raycast.app/Contents/Resources/RaycastCommands_RaycastCommands.bundle/Contents/Resources/api/node_modules/@raycast/api/index.js:24:10558)
    at al (/Applications/Raycast.app/Contents/Resources/RaycastCommands_RaycastCommands.bundle/Contents/Resources/api/node_modules/@raycast/api/index.js:24:10663)
    at Ts (/Applications/Raycast.app/Contents/Resources/RaycastCommands_RaycastCommands.bundle/Contents/Resources/api/node_modules/@raycast/api/index.js:24:9779)
    at QO (/Applications/Raycast.app/Contents/Resources/RaycastCommands_RaycastCommands.bundle/Contents/Resources/api/node_modules/@raycast/api/index.js:24:8740)
    at ray-root
    at Ef (/Applications/Raycast.app/Contents/Resources/RaycastCommands_RaycastCommands.bundle/Contents/Resources/api/node_modules/@raycast/api/index.js:24:10083)
 ```
 
Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions regarding resolving this issue.

**Comment by selfish** [comment]
Will do ASAP

**Comment by pernielsentikaer** [comment]
Awesome @selfish 💪  — Just a quick question, is the only for 1password 7, or will it work with 1password 8 too?

**Comment by selfish** [comment]
@pernielsentikaer error resolved.

As for your question, there's not much documentation for 1pass 8 for now, but I tend to expect full backward compatibility.
Are you hinting at updating the name to not be v7 specific? I don't mind including that. Should be real quick to do.

**Comment by selfish** [comment]
Okay, scratch that. I just updated and for now, 1pass 8 won't offer the 3rd party integrations option. So yeah, only 1p 7 for now.

**Review by pernielsentikaer — APPROVED** [review]
Awesome @selfish 💪 

All things I found has been resolved, I'm going to accept it 💯

**Pull request merged**